src/http.cc

```diff
@@ -1255,6 +1255,8 @@ HttpStateData::readReply(const CommIoCbParams &io)
         debugs(11, 2, io.conn << ": read failure: " << xstrerr(rd.xerrno));
         const auto err = new ErrorState(ERR_READ_ERROR, Http::scBadGateway, fwd->request, fwd->al);
         err->xerrno = rd.xerrno;
+        static const auto d = MakeNamedErrorDetail("SRV_READ_FAILURE");
+        err->detailError(d);
```

Comment:
This change was intended to help differentiate between ERR_READ_ERROR cases, particularly this case and our EOF case, thus avoiding default and non-informative 'WITH_SERVER'. I agree that we s should probably adopt another (more complex) approach, e.g., supplying errno to SysErrorDetail (and probably create MakeNamedErrorDetail if errno is missing).  So I do not mind removing  these lines for now, since our 'SRV_PREMATURE_EOF' detail should be enough to distinguish the EOF case from another cases (including default ones).

>  ErrorState::BuildHttpReply() will add this detail to the request. We are probably copying this detail to ALE.

I am not sure that I understood the problem you are talking about. Note that ErrorState::BuildHttpReply() updates the Request error with HttpRequest::detailError() - so it will 'add' the detail only at the first call (and ignore the second) - that is in line with Request::error documentation ("the first transaction problem..."). For logging we use AccessLogEntry::error(), that returns either ALE's error or HttpRequest's error - and not the combination of them.

Reply:
>> ErrorState::BuildHttpReply() will add this detail to the request. We are probably copying this detail to ALE.

> I am not sure that I understood the problem you are talking about.

The same error should not be detailed in two places (request and ALE), especially since multiple error details should be (and eventually will be) accumulated/concatenated rather than overwritten or ignored like they are now.

> AccessLogEntry::error() ... returns either ALE's error or HttpRequest's error - and not the combination of them

You are explaining why the current code works correctly, but I am not claiming that the current PR code does not "work". I am only pointing out a conceptual problem with this PR change -- duplicating error detail.

It sounds like we agree that these (arguably problematic and possibly difficult to complete) changes are not needed in this PR. Let's remove them and deal with detailing this error in a dedicated project, addressing both problems.

Reply:
> Let's remove them

Ok.